### DISCUSSION TITLE
Raspberry Pi GPIO Pin control

### DISCUSSION THREAD

**Discussion by DarkFiguringitout** [kormax/apple-home-key-reader#16] [Ideas]
Is there a way to control the GPIO pins of a raspberry pi when unlocking? For example, when unlocked it can trigger a specific pin which can be used in automations that are in close proximity.

Right now, I have a bluetooth lock (Bold smart lock), currently use a ESP8266 to bridge the contacts to activate the remote to unlock the lock. Would be nice to keep the scanner + key inside behind the door glass, and trigger the key directly without an internet connection, whenever I want to unlock the door.

**Comment by kormax** [comment]
Yes, entirely possible. 

You can install RPi.GPIO or gpiozero  library, and add that GPIO-triggering logic inside of `on_endpoint_authenticated` method, which can be found in `accessory.py`.

**Comment by dlbogdan** [comment]
what about the other way around? set the status of the lock (resync the status) externally ? 
example: 
the bluetooth smart lock has its own control buttons which when activated, it would not be in sync anymore with homekey .

**Reply by dlbogdan** [reply]
nevermind I've figured it out. Going to use set_lock_target_state and on_endpoint_authenticated to actually set the real lock ( eq-3 bluetooth smart lock) and will also check the real bluetooth lock state in get_lock_current_state and if it's not the same, will just call set_lock_target_state on the new state. [ ?? or just set the variable? need to test things here ]

I think this should be enough for a tightly synced system, right ?
